[WIP] Gui refactoring

Illithidek opened the pull request.

Commit:
Refactoring

-Delete useless comments
-Move things to initializer list

Commit:
GUI Refactoring

-Rename CIntObject to View

Commit:
GUI Refactoring

-Move View to subfolder view

Commit:
GUI Refactoring

-Initialize file for ViewGroup
-Init getter and setter for View parent

alexvins commented on client/widgets/Buttons.cpp:
```diff
@@ -145,14 +145,20 @@ void CButton::onButtonClicked()
+		
 		logAnim->trace("%s%s at %dx%d", prefix, typeid(*parent).name(), parent->pos.x, parent->pos.y);
-		parent = parent->parent;
+		parent = getParent()->getParent();
```
should be `parent->getParent()`

FeniksFire replied:
Thanks for noticing!

alexvins commented on client/gui/view/View.h:
```diff
@@ -67,7 +67,10 @@ class View : public IShowActivatable
-
+	
+	View *getParent() const;
+	void setParent(View *parent);
```
This should be removed, or made protected it you really want to have setter anyway.
You should use `addChild` `removeChild` from outside

FeniksFire replied:
I will. Give me some time, I just want rework all these things, so I will made some things temporary :).

Commit:
GUI Refactoring

-Get rid with parent_m

Commit:
GUI Refactoring

-Get rid of code dublication

Commit:
GUI Refactoring

-GuiHandler method that managing event now take pointer to event

Commit:
GUI Refactoring

-Get rid of GH.current event usage

Commit:
GUI Refactoring

-Get rid of active_m in View

Commit:
GUI Refactoring

-Move part of the code outside View

Commit:
GUI Refactoring

-Move part of the code outside View

Commit:
GUI Refactoring

-Move handling click event from GuiHandler To View
-Connection of the processed clickable elements to the one clickable list

Commit:
GUI Refactoring

-Remove code duplication

Commit:
Get rid of mouseState in View
